pkg/executor/aggfuncs/spill_serialize_helper.go

```diff
@@ -0,0 +1,252 @@
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeBool(value.isNull, s.buf)
+	s.buf = spill.SerializeDuration(value.val.Duration, s.buf)
+	return spill.SerializeInt(value.val.Fsp, s.buf)
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4MaxMinString(value partialResult4MaxMinString) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeBool(value.isNull, s.buf)
+
+	// Do not return s.buf directly as we need to enlarge it when serialized data is large
+	s.buf = append(s.buf, value.val...)
+	return s.buf
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4MaxMinJSON(value partialResult4MaxMinJSON) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeBool(value.isNull, s.buf)
+
+	// Do not return s.buf directly as we need to enlarge it when serialized data is large
+	s.buf = spill.SerializeBinaryJSON(&value.val, s.buf)
+	return s.buf
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4MaxMinEnum(value partialResult4MaxMinEnum) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeBool(value.isNull, s.buf)
+
+	// Do not return s.buf directly as we need to enlarge it when serialized data is large
+	s.buf = spill.SerializeEnum(&value.val, s.buf)
+	return s.buf
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4MaxMinSet(value partialResult4MaxMinSet) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeBool(value.isNull, s.buf)
+
+	// Do not return s.buf directly as we need to enlarge it when serialized data is large
+	s.buf = spill.SerializeSet(&value.val, s.buf)
+	return s.buf
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4AvgDecimal(value partialResult4AvgDecimal) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeMyDecimal(&value.sum, s.buf)
+	return spill.SerializeInt64(value.count, s.buf)
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4AvgFloat64(value partialResult4AvgFloat64) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeFloat64(value.sum, s.buf)
+	return spill.SerializeInt64(value.count, s.buf)
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4SumDecimal(value partialResult4SumDecimal) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeMyDecimal(&value.val, s.buf)
+	return spill.SerializeInt64(value.notNullRowCount, s.buf)
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4SumFloat64(value partialResult4SumFloat64) []byte {
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeFloat64(value.val, s.buf)
+	return spill.SerializeInt64(value.notNullRowCount, s.buf)
+}
+
+func (s *SpillSerializeHelper) serializeBasePartialResult4GroupConcat(value basePartialResult4GroupConcat) []byte {
+	valsBuf := value.valsBuf.Bytes()
+	valsBufLen := int64(len(valsBuf))
+	buffer := value.buffer.Bytes()
+
+	s.buf = s.buf[:0]
+	s.buf = spill.SerializeInt64(valsBufLen, s.buf)
+	s.buf = append(s.buf, valsBuf...)
+
+	// Do not return s.buf directly as we need to enlarge it when serialized data is large
```

Comment:
> What does this comment mean?

Because I want to remind other that we shouldn't merge the last two clause into `return append(s.buf, buffer...)`

Reply:
OK, but this reminds me that all the `serializexxx4xxx` can be defined as no return value since the caller can get `s.buf` directly after if he wants to get the buf?